compiler/src/dotty/tools/dotc/transform/PatternMatcher.scala

```diff
@@ -335,10 +335,16 @@ object PatternMatcher {
               patternPlan(casted, pat, onSuccess)
             })
         case UnApply(extractor, implicits, args) =>
-          val mt @ MethodType(_) = extractor.tpe.widen
-          var unapp = extractor.appliedTo(ref(scrutinee).ensureConforms(mt.paramInfos.head))
-          if (implicits.nonEmpty) unapp = unapp.appliedToArgs(implicits)
-          val unappPlan = unapplyPlan(unapp, args)
+          val unappPlan = if (!defn.isSubtypeOfBottom(scrutinee.info)) {
```

Comment:
I'd turn the branches around so that the condition can be positive.

Reply:
Done